app/api/suggestions/routes.ts

```diff
@@ -0,0 +1,124 @@
+import { Application } from 'express';
+import { Suggestions } from 'api/suggestions/suggestions';
+import { InformationExtraction } from 'api/services/informationextraction/InformationExtraction';
+import { validateAndCoerceRequest } from 'api/utils/validateRequest';
+import { needsAuthorization } from 'api/auth';
+import { config } from 'api/config';
+import { parseQuery } from 'api/utils/parseQueryMiddleware';
+import { IXSuggestionsQuerySchema } from 'shared/types/suggestionSchema';
+import { objectIdSchema } from 'shared/types/commonSchemas';
+import { serviceMiddleware } from './serviceMiddleware';
+
+let IX: InformationExtraction;
+if (config.externalServices) {
+  IX = new InformationExtraction();
+}
+
+export const suggestionsRoutes = (app: Application) => {
+  app.get(
+    '/api/suggestions/',
+    serviceMiddleware,
+    needsAuthorization(['admin']),
+    parseQuery,
+    validateAndCoerceRequest({
+      properties: {
+        query: IXSuggestionsQuerySchema,
+      },
+    }),
+    async (req, res, _next) => {
+      const suggestionsList = await Suggestions.get(
+        { language: req.language, ...req.query.filter },
+        { page: req.query.page }
+      );
+      res.json(suggestionsList);
+    }
+  );
+
+  app.post(
+    '/api/suggestions/train',
+    serviceMiddleware,
+    needsAuthorization(['admin']),
+    validateAndCoerceRequest({
+      properties: {
```

Comment:
Same as starting on line 69, maybe refactor in this file.

Reply:
Actually, it's different, the first one is expecting a value into the body while the second one is expecting the value into the query. I extracted it into a function that receives this root option.